libs/@local/harpc/crates/wire/src/codec/decode.rs

```diff
@@ -0,0 +1,173 @@
+use std::io;
+
+use bytes::Bytes;
+use error_stack::{Context, Result};
+use graph_types::account::AccountId;
+use tokio::io::{AsyncRead, AsyncReadExt};
+use uuid::Uuid;
+
+pub trait DecodePure: Send + Sync + Sized {
```

Comment:
It doesn't take a context, tbh only reason it is named `Pure` is because I couldn't come up with a better name. Any suggestion is very much appreciated!

Reply:
Can't we just use `Decode`? 😄